fetch.ts

```
@@ -150,6 +148,46 @@ function isCloudlfareProtectionKnown(hostname: string) {
     return isKnown;
 }
 
+// Octokit instance injected by caller (optional). We avoid hard dependency so library users aren't forced to provide it.
+let _octokit: unknown | undefined;
+export function setOctokit(octokit: unknown) {
+    _octokit = octokit;
+}
+
+async function getGithubIssueCommentAnchors(owner: string, repo: string, issueNumber: number) {
+    if (
+        _octokit == null || typeof _octokit !== "object" || _octokit === null ||
+        typeof (_octokit as { request?: unknown }).request !== "function"
+    ) {
+        return { anchors: new Set<string>(), comments: new Set<string>() };
```

We should probably throw an error here instead of just returning no anchors

---

```suggestion
        throw new TypeError("setOctokit has not been called with a valid value");
```